specs/deneb/beacon-chain.md

```diff
@@ -187,6 +187,19 @@ def kzg_commitment_to_versioned_hash(kzg_commitment: KZGCommitment) -> Versioned
     return VERSIONED_HASH_VERSION_KZG + hash(kzg_commitment)[1:]
 ```
 
+#### `is_valid_merkle_path`
+
+```python
+def is_valid_merkle_path(leaf: Bytes32, branch: Sequence[Bytes32], gindex: int, root: Root) -> bool:
```

Comment:
We have 
```ts
def is_valid_merkle_branch(leaf: Bytes32, branch: Sequence[Bytes32], depth: uint64, index: uint64, root: Root) -> bool:
```
We could convert the gindex of this PR's proof into a (depth, index) tuple or generalize `is_valid_merkle_branch` to accept gindexes

Reply:
I think its better to use [`is_valid_merkle_branch`](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#is_valid_merkle_branch).

The corresponding client code is widely supported and well-tested and so there is less to get correct for this change (that seems important and also is a bit under time pressure).

Reply:
+1 on reusing `is_valid_merkle_branch` rather than creating new and similar helpers.
To use `is_valid_merkle_branch, we may reuse light-client helper `get_subtree_index`.

https://github.com/ethereum/consensus-specs/blob/612d1485ba432ca7d8fcf3f34a9b9e5dcaa47900/specs/altair/light-client/sync-protocol.md?plain=1#L260-L265

I changed it in https://github.com/ethereum/consensus-specs/pull/3531/commits/b7e0b88cb18a9e726af9ff1e7828c778de899d77, and then the `test_blob_sidecar_inclusion_proof_correct` test passed.

If we agree to do this way, we need to do some more clean-ups:
1. Delete `is_valid_merkle_path`
2. Add a note that `get_subtree_index` helper is from light client specs.

Reply:
If `get_subtree_index` is not specific to the light client, should we move it out of the light client specs?

Reply:
Even if we decide to keep `is_valid_merkle_path`, I think I cannot tell the difference between `is_valid_merkle_path` and `is_valid_merkle_branch` from the names.

Probably it's better to change one of the names or both of them.

Reply:
I totally agree with https://github.com/ethereum/consensus-specs/commit/b7e0b88cb18a9e726af9ff1e7828c778de899d77. I think even if we assume that `is_valid_merkle_branch` takes the non-generalized indexes, it can be reused for the generalized indexes as well since the code looks very similar.